### ISSUE TITLE
Less layers freezed

### ISSUE THREAD

**Issue by ShiningCoding** [matterport/Mask_RCNN#364]
Thank you for your code! @waleedka 
To make it run for my own project, I have modified the code in these ways:
1. Using python 2.7, modify some functions related to download or other incompatible aspects.
2. Move the layers related to segmentation, as well as the mask related part in the input generators.
Then I run the code, and it can runs, the losses are all decreasing seemingly normally. But:
1. I noticed the losses decrease slowly after 10+ epochs in the first stage. So I stopped it and made it begin to run the second stage (layer 4+ trainable), but the total loss dramatically increased at the first epoch, compared with that before stopped in the first stage.
2. I used the trained model to detect in some sample images, and no right objections can be detected.

My dataset has 21 classes , each with ~2k images for training. The config is the same as that in your code for training coco. 

Can you give some advice for make the model work as for my project?

Thank you!

**Comment by nhanteh** [comment]
Hi,

I have a similar problem with similar number of classes and images. I can never get it to detect any object correctly and the loss decreases slowly. 

Thanks.

**Comment by ShiningCoding** [comment]
@nhanteh Maybe you can try increase the learning rate a bit?  Or check the IMAGE_MIN_DIM/IMAGE_MAX_DIM/RPN_ANCHOR_SCALES to match your training images size better? I think you should make the model scan the whole training images at least several times and the  loss decrease significantly first, then check the detection results.

Good luck!

**Comment by waleedka** [comment]
@ShiningCoding It's normal for the loss to drop fast in the beginning and then slow down. It's also normal for the loss to go up a little bit for a short time when you change the training parameters. But you said it went up significantly, so probably there is something wrong. Here are a few things you might want to try:

1. Run the detection on the model trained with the first stage only. You should get some correct detections. If you don't then the issue is happening early on, and is not related to the transition to stage 2 of training.
2. Use the inspect_data notebook to verify that your data is being loaded correctly.
3. Use the inspect_model notebook and focus on the part that runs detection one step at a time. That'll help isolate where in the pipeline the problem is happening.

**Comment by ShiningCoding** [comment]
@waleedka Thank you very much. I have found the problem. It's due to the data augmentation and detection rectangles loading. Thanks again for your advice and your detail examples for the beginners.

**Comment by nhanteh** [comment]
@waleedka Thanks for the information, when I inspect_data everything seems fine and I have all the classes and bboxes presented. However, when I inspect the first stage of the model it seems like I only detect one class for all bboxes. I appreciate any suggestion...
